### DISCUSSION TITLE
Data at rest on a Server.

### DISCUSSION THREAD

**Discussion by bits0rcerer** [padloc/padloc#413] [Q&A] [answered]
### I am wondering how Padloc is storing persistent data on the server. 

1. Is there a single moment where a password is in plain text somewhere in memory or on disk?
2. Whats the case with the pwa?

Sadly the part [Compromised Server](https://github.com/padloc/padloc/blob/master/security.md#compromised-server) is not done yet.

It would be awesome if someone can provide an answer or guide me to the right place in the code.

**Comment by MaKleSoft** [comment, marked as answer]
Hi there! Padloc is designed with a zero-knowledge architecture, meaning no sensitive data is ever exposed in plaintext to our servers or the network. In fact, all data is fully end-to-end encrypted and never leaves your device without multiple layers of protection. On the client side, all sensitive data is encrypted at rest as well, so the only time your data ever exists in plain text is in memory on the client side, and only while the app is running and unlocked.

I hope that answers your question! Let me know if you'd like to hear more details.
